src/en/guide/basics/routing.md

```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
 This gives you the freedom to define specific matching patterns for your use case. 
 
 In the example shown, we are looking for a date that is in `YYYY-MM-DD` format.
+
+::: tip Advanced
+
+But for complex routing, the above example is still too simple.
+
+Sometimes you want to match on a part of a path segment:
+
+```text
+/image/123456789.jpg
+```
+
+If you wanted to match the file pattern, but only capture the numeric portion, you need to do some regex fun:
+
+```python
+app.route(r"/image/<img_id:(?P<img_id>\d+)\.jpg>")
+```
+
+Further, these should all be acceptable:
+
+```python
+@app.get(r"/<foo:[a-z]{3}.txt>")                # matching on the full pattern
+@app.get(r"/<foo:([a-z]{3}).txt>")              # defining a single matching group
+@app.get(r"/<foo:(?P<foo>[a-z]{3}).txt>")       # defining a single named matching group
+@app.get(r"/<foo:(?P<foo>[a-z]{3}).(?:txt)>")   # defining a single named matching group, with one or more non-matching groups
```

Comment:
I think we should make all of these examples use the `\d+` with `jpg` pattern for consistency with the first example we give.

Maybe the last one can be an exception and be a little different:

```python
r"/<foo:(?P<foo>\d+).(?:jpg|png)>
```